File: Iteration2/code/CSE3063GRP12/src/core/models/concretes/Lecturer.java
package core.models.concretes;

import java.util.ArrayList; 
import core.models.abstracts.User;

public class Lecturer extends User {
    private ArrayList<Course> listOfCoursesGiven;

    public Lecturer(String id, String firstName, String lastName, String userName,
            String password, ArrayList<Course> listOfCoursesGiven) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.userName = userName;
        this.password = password;
        this.listOfCoursesGiven = listOfCoursesGiven;
    }

    public void setListOfCoursesGiven(ArrayList<Course> listOfCoursesGiven) {
        this.listOfCoursesGiven = listOfCoursesGiven;
    }

    public ArrayList<Course> getListOfCoursesGiven() {
        return listOfCoursesGiven;
    }

}

File: Iteration3/code/CSE3063GRP12/src/core/models/concretes/Lecturer.py
from core.models.abstracts.User import User

class Lecturer(User):
    def __init__(self, id=None, firstName=None, lastName=None, userName=None, password=None, listOfCoursesGiven=None):
        super().__init__(id, firstName, lastName, userName, password)
        self.listOfCoursesGiven = listOfCoursesGiven if listOfCoursesGiven else []

    def set_list_of_courses_given(self, listOfCoursesGiven):
        self.listOfCoursesGiven = listOfCoursesGiven

    def get_list_of_courses_given(self):
        return self.listOfCoursesGiven
    
    def to_dict(self):
        return {
            "id": self.id,
            "firstName": self.firstName,
            "lastName": self.lastName,
            "userName": self.userName,
            "password": self.password,
            "listOfCoursesGiven": [course.to_dict() if hasattr(course, 'to_dict') else course for course in self.listOfCoursesGiven]
        }